WallsAndHoles/newmapdialog.cpp

```diff
@@ -17,4 +19,39 @@ NewMapDialog::~NewMapDialog()
 void NewMapDialog::on_buttonBox_accepted()
 {
     result = NewMapData(ui->widthSpinBox->value(), ui->heightSpinBox->value());
+
+    oneChecked = ui->checkBox_Ceiling->isChecked()||ui->checkBox_Default->isChecked()||
+            ui->checkBox_Indoor->isChecked()||ui->checkBox_Outdoor->isChecked();
+}
+
+void NewMapDialog::on_checkBox_Default_toggled(bool checked)
+{
+    if(checked)
```

Comment:
`TileMapProperty.defaultMap = checked;`
Same for bellow, In fact, to keep things compact, you can just make this a lambda:
```
connect(checkbox, &CheckBox::toggled,
               this, [this](bool checked){ TileProperty.defalutMap = checked; });
```

(Of course this is for when you move to hard coding the UI)